cocos/platform/CCFileUtils.cpp

```diff
@@ -603,7 +641,9 @@ bool FileUtils::init()
 
 void FileUtils::purgeCachedEntries()
 {
+    _cacheLock.writeLock();
     _fullPathCache.clear();
+    _cacheLock.writeUnlock();
```

Comment:
Do you think it would be worthwhile to write a lock_guard equivalent for writeLock and readLock that uses RAII? Something like 

`WriteLockGaurd lock(_cacheLock); ReadLockGaurd lock(_cacheLock);` ?